packages/slate/src/models/node.js

```diff
@@ -1987,6 +1987,22 @@ class Node {
   validate(schema) {
     return schema.validateNode(this)
   }
+
+  /**
+   * The first descendant key requiring validation
+   *
+   * @param {Schema} schema
+   * @return {Node|Text|Null}
+   */
+
+  getFirstInvalidDescendant(schema) {
+    let result = null
+    this.nodes.find(n => {
+      result = n.validate(schema) ? n : n.getFirstInvalidDescendant(schema)
+      return result
+    })
+    return result
+  }
```

Comment:
```js
getFirstInvalidDescendant(schema) {
  return this.nodes.find(
    n => n.validate(schema)
      ? n
      : n.getFirstInvalidDescendant(schema)
  ) || null;
}
```

Reply:
I prefer the original, that suggested version is hard to read, and very hard to set breakpoints in when debugging why something is going wrong.

Reply:
@ianstormtaylor By 'the original', do you mean the original one that 
```
while(node && child) {
   ....
  child = node.nodes.find(n =>....);
}
```

The only thing I am worrying about the original one is that, if this. getFirstInvalidDescendant can return itself, then the this.__cache.getFirstInvalidDescendant will have a refer to itself.  Shall we allow cycle referring in the code? (Or just trust browsers can do the cycle detection?)

Reply:
@zhujinxuan I'm pretty sure @ianstormtaylor just meant to ignore my comment here, what you had is correct.

Reply:
I did not notice the file name.  I have just removed my unnecessary commits.